astropy/visualization/wcsaxes/axislabels.py

```diff
@@ -0,0 +1,126 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+
+from __future__ import print_function, division, absolute_import
+
+import numpy as np
+
+from matplotlib.text import Text
+import matplotlib.transforms as mtransforms
+
+from .frame import RectangularFrame
+
+
+class AxisLabels(Text):
+
+    def __init__(self, frame, minpad=1, *args, **kwargs):
+        self._frame = frame
+        super(AxisLabels, self).__init__(*args, **kwargs)
+        self.set_clip_on(True)
+        self.set_visible_axes('all')
+        self.set_ha('center')
+        self.set_va('center')
+        self._minpad = minpad
+
+    def get_minpad(self, axis):
+        try:
+            return self._minpad[axis]
+        except TypeError:
+            return self._minpad
+
+    def set_visible_axes(self, visible_axes):
```

Comment:
Is the structure with `set_*` and `get_*` functions matplotlib-specific? It would be more pythonic to have properties...
(obviously, not at all relevant for the question of whether or not this can be merged).